tests/core/pyspec/eth2spec/test/bellatrix/confirmation_rule/test_confirmation_rule.py

```diff
@@ -0,0 +1,671 @@
+from eth2spec.test.context import spec_state_test, with_config_overrides, with_presets, with_bellatrix_and_later
+from eth2spec.test.helpers.attestations import (
+    get_valid_attestation_at_slot,
+    next_epoch_with_attestations,
+    next_slots_with_attestations,
+)
+from eth2spec.test.helpers.constants import MINIMAL
+from eth2spec.test.helpers.fork_choice import (
+    add_block,
+    get_genesis_forkchoice_store_and_block,
+)
+
+from eth2spec.test.helpers.state import (
+    next_epoch,
+)
+
+
+def on_tick_step(spec, store, time, test_steps):
+    spec.on_tick(store, time)
+    test_steps.append({"tick": int(time)})
+
+
+def tick_to_next_slot(spec, store, test_steps):
+    time = store.genesis_time + (spec.get_current_slot(store) + 1) * spec.config.SECONDS_PER_SLOT
+    on_tick_step(spec, store, time, test_steps)
+
+
+def tick_and_add_block(
+    spec, store, signed_block, test_steps, valid=True, merge_block=False, block_not_found=False, is_optimistic=False
+):
+    pre_state = store.block_states[signed_block.message.parent_root]
+    if merge_block:
+        assert spec.is_merge_transition_block(pre_state, signed_block.message.body)
+
+    block_time = pre_state.genesis_time + signed_block.message.slot * spec.config.SECONDS_PER_SLOT
+    while store.time < block_time:
+        time = pre_state.genesis_time + (spec.get_current_slot(store) + 1) * spec.config.SECONDS_PER_SLOT
+        on_tick_step(spec, store, time, test_steps)
```

Comment:
Is `on_tick_and_append_step` v.s. `on_tick_step` here the only difference between this `tick_and_add_block` and the helper from `helpers/fork_choice.py`? Could we refactor it to reuse one common helper?